### PR TITLE
firmwareLinuxNonfree -> linux-firmware

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change

This renames our `firmwareLinuxNonfree` package to `linux-firmware`.
There is prior art for this in multiple other distros[1] [2] [3].

Besides making the package more discoverable by those searching for the
usual name, this also brings it in-line with the `kebab-case` we
normally see in `nixpkgs` pnames, and removes the `Nonfree` information
from the name, which I consider redundant given it's present in
`meta.license`.

The corresponding alias has been added, so this shouldn't break
anything.

[1]: https://archlinux.org/packages/core/any/linux-firmware/
[2]: https://src.fedoraproject.org/rpms/linux-firmware
[3]: https://packages.gentoo.org/packages/sys-kernel/linux-firmware

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by grahamc** [comment]
This looks like a good idea to me.

**Review by raboof — APPROVED** [review]
I think this is a nice improvement, especially since not *all* the firmware modules in this package are, in fact, nonfree, making this arguably a more accurate/less confusing name.

File: nixos/modules/hardware/all-firmware.nix [hunk 1]
```diff
@@ -31,7 +31,7 @@ in {
       type = types.bool;
       description = ''
         Turn on this option if you want to enable all the firmware with a license allowing redistribution.
-        (i.e. free firmware and <literal>firmware-linux-nonfree</literal>)
+        (i.e. free firmware and <literal>linux-firmware</literal>)
```
[the remark this answers is not in the document]
Comment (lovesegfault) [reply nixos/modules/hardware/all-firmware.nix:34]: Reworded, see what you think of it now :)

**Review by pennae — APPROVED** [review]

File: nixos/modules/hardware/all-firmware.nix [hunk 2]
```diff
@@ -31,7 +31,7 @@ in {
       type = types.bool;
       description = ''
         Turn on this option if you want to enable all the firmware with a license allowing redistribution.
-        (i.e. free firmware and <literal>firmware-linux-nonfree</literal>)
+        (i.e. free and non-free firmware contained in <literal>linux-firmware</literal>)
```
Comment (raboof) [inline nixos/modules/hardware/all-firmware.nix:34]: Hmm, this sounds confusing, as this option will enable not *just* the (free and non-free) firmware from `linux-firmware`, but also some others (`intel2200BGFirmware`, `rtl8192su-firmware`, `rt5677-firmware`, ...).

I don't feel strongly about this, so feel free to ignore, but I still think just removing this line would be clearest ;)

Comment (lovesegfault) [reply nixos/modules/hardware/all-firmware.nix]: Done :)

**Pull request merged**